### PR TITLE
Update composer dependencies and unit tests

### PR DESCRIPTION
**Pull request by drpayyne**
# Description

- updated php to `^7.3||^8.0`
- updated phpunit/phpunit to `^9.5`
- updated nikic/php-parser to `^4.10`
- updated phpstan/phpdoc-parser to `^0.5`
- replaced zendframework/zend-stdlib with laminas/laminas-stdlib as Zend packages are deprecated
- sorted composer dependencies :)
- updated php to `7.3` in `.github/workflows/php.yml`

## Fixed Issues

1. Fixes https://github.com/magento/magento2/issues/32872

## Contribution checklist (*)

- [x] Pull request has a meaningful description of its purpose
- [x] All commits are accompanied by meaningful commit messages
- [x] All new or changed code is covered with unit/integration tests (if applicable)
- [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by fascinosum** [comment]
hi @drpayyne,
thank you for your contribution. 

Could you redirect it to the develop branch

You have a failed test build because PHP 7.2 is out of support. Could you update the [PHP version](https://github.com/magento/magento-semver/blob/9ef11a45f83fb7fc25badea59d0441ddc771fc61/.github/workflows/php.yml#L18) used for the build? We need to run build with PHP 7.3

To properly test these changes, we need to create a temporary PR into [magento2-infrastructure](https://github.com/magento/partners-magento2-infrastructure) and update [composer.json dependency](https://github.com/magento/partners-magento2-infrastructure/blob/1e52186c90a727a3432aecd7514c6c14965462b6/composer.json#L24) to use your branch as a source. See [an example](https://github.com/magento/partners-magento2-infrastructure/commit/823fd5c053b4767d6359b6ae05450a79d8a272a2#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34) 
We need to run the SVC build with PHP 7.3 and PHP 7.4 for that PR
cc: @andrewbess

**Comment by drpayyne** [comment]
Hi @fascinosum, I've updated PHP in composer and the GitHub workflow config file. Unfortunately I don't have access to the infra repo you have linked. I am part of a partner org (EY), so I'm not sure if that repo is available to partners.

**Review by andrewbess — APPROVED** [review]
Hello @drpayyne 
Thank you for your contribution.
I fixed the infrastructure and code styling problems.
Changes are approved.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @drpayyne,
Could you review and fix the warnings when running unit tests? 
You can find them here:
https://github.com/magento/magento-semver/pull/61/checks
image

**Comment by drpayyne** [comment]
Sure @ihor-sviziev, I will fix those today.

File: composer.json [hunk 1]
```diff
@@ -8,18 +8,18 @@
...
         "tomzx/php-semver-checker": "^0.14.0",
-        "wikimedia/less.php": "~1.8.0",
-        "zendframework/zend-stdlib": "^3.2.1",
-        "nikic/php-parser": "~4.4.0||~4.5.0||~4.6.0",
-        "sabre/xml": "^2.1"
+        "wikimedia/less.php": "~1.8.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^6.5.0",
+        "phpunit/phpunit": "9.3.0",
```
Comment (ihor-sviziev) [inline composer.json:22]: Can we allow installing 9.3.x?

Comment (andrewbess) [reply composer.json]: Hello @ihor-sviziev.
I changed it temporary to have some checks.
Now, I fixed all changes and squashed commits.

**Review by ihor-sviziev — APPROVED** [review]

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Please update recent changes from the develop branch

**Comment by andrewbess** [comment]
This pull request (PR) closed because has been created from wrong branch (master instead of develop) and has a lot of merge-conflicts.
New PR is https://github.com/magento/magento-semver/pull/65

**Pull request closed without merging**